src/lightning_lite/connector.py

```diff
@@ -0,0 +1,596 @@
+    CheckpointIO,
+    DeepSpeedPrecision,
+    NativeMixedPrecision,
+    Precision,
+    TPUBf16Precision,
+    TPUPrecision,
+)
+from lightning_lite.plugins.environments import (
+    ClusterEnvironment,
+    KubeflowEnvironment,
+    LightningEnvironment,
+    LSFEnvironment,
+    SLURMEnvironment,
+    TorchElasticEnvironment,
+)
+from lightning_lite.plugins.precision.double import DoublePrecision
+from lightning_lite.strategies import (
+    DDPShardedStrategy,
+    DDPSpawnShardedStrategy,
+    DDPSpawnStrategy,
+    DDPStrategy,
+    DeepSpeedStrategy,
+    SingleDeviceStrategy,
+    SingleTPUStrategy,
+    Strategy,
+    STRATEGY_REGISTRY,
+    TPUSpawnStrategy,
+)
+from lightning_lite.strategies.ddp_spawn import _DDP_FORK_ALIASES
+from lightning_lite.utilities import _StrategyType, device_parser, rank_zero_deprecation, rank_zero_info, rank_zero_warn
+from lightning_lite.utilities.imports import _HPU_AVAILABLE, _IPU_AVAILABLE, _IS_INTERACTIVE, _TPU_AVAILABLE
+
+_PLUGIN = Union[Strategy, Precision, ClusterEnvironment, CheckpointIO]
+_PLUGIN_INPUT = Union[_PLUGIN, str]
+
+
+class _Connector:
+    """The Connector parses several Lite arguments and instantiates the Strategy including other components such as
+    the Accelerator and Precision plugins.
+
+        A. accelerator flag could be:
+            1. accelerator class
+            2. accelerator str
+            3. accelerator auto
+
+        B. strategy flag could be :
+            1. strategy class
+            2. strategy str registered with STRATEGY_REGISTRY
+            3. strategy str in _strategy_type enum which listed in each strategy as
+               backend (registed these too, and _strategy_type could be deprecated)
+
+        C. plugins flag could be:
+            1. List of str, which could contain:
+                i. precision str (Not supported in the old accelerator_connector version)
+                ii. checkpoint_io str (Not supported in the old accelerator_connector version)
+                iii. cluster_environment str (Not supported in the old accelerator_connector version)
+            2. List of class, which could contains:
+                i. precision class (should be removed, and precision flag should allow user pass classes)
+                ii. checkpoint_io class
+                iii. cluster_environment class
+
+
+    priorities which to take when:
+        A. Class > str
+        B. Strategy > Accelerator/precision/plugins
+    """
+
+    def __init__(
+        self,
+        accelerator: Optional[Union[str, Accelerator]] = None,
+        strategy: Optional[Union[str, Strategy]] = None,
+        devices: Optional[Union[List[int], str, int]] = None,
+        num_nodes: int = 1,
+        precision: Union[int, str] = 32,
+        plugins: Optional[Union[_PLUGIN_INPUT, List[_PLUGIN_INPUT]]] = None,
+        tpu_cores: Optional[Union[List[int], str, int]] = None,  # deprecated
+        gpus: Optional[Union[List[int], str, int]] = None,  # deprecated
```

Comment:
do we really want to carry over this deprecation? Having a newly designed package would be the perfect time of getting rid of stuff like this.

Reply:
It would be nice not to, but it was already in Lite beforehand. So I see two options:

A) In the new package, delete these. The Lite class that will stay in PL for backward compatibility will have additional logic to handle this (using the old connector?). 
B) We carry it over but shorten the deprecation period from 2.0 to something earlier (is this breaking our promises??)

Reply:
I would go for A), have a tiny wrapper that basically only translates `tpu_cores=X` to `accelerator='tpu', devices=X` and `gpus=X` to `accelerator='gpu', devices=X`. This should be a minimal helper which could then be reused for PL

Reply:
Waiting to implement this in case other reviewers want to comment first.